### DISCUSSION TITLE
Suggested Bug Fixes in PeakRDL code

### DISCUSSION THREAD

**Discussion by arhum-alam** [SystemRDL/systemrdl-compiler#179] [Ideas]
Hi developers,

We have made use of the systemrdl-compiler and PeakRDL generator codes within our company to generate RTL, UVM, C++ and HTML outputs. In the process, we have identified a couple of bug fixes that other users may have faced as well. What would you suggest is the best way to report these fixes?

Arhum

---
_Arhum Alam
Design Engineer at AccelerComm Ltd.
Southampton, United Kingdom_

**Comment by amykyta3** [comment]
Hi Arhum,
Thanks for your enthusiasm and willingness to contribute to the SystemRDL/PeakRDL projects!

In general, bug fixes should always start by submitting an issue ticket to the appropriate project. If you are able to submit a pull request, that is great too, but it is always best to discuss the change with us first.

Please review the [contribution guidelines for the systemrdl-compiler project](https://github.com/SystemRDL/systemrdl-compiler/blob/main/CONTRIBUTING.md).
Many of these guidelines also apply equally to the other PeakRDL projects.

**Reply by arhum-alam** [reply]
Hi Alex,

Thanks for your reply. I'll go ahead and submit issue tickets to highlight the potential bugs we have come across. If it is indeed identified as a bug in the generator, I'll open a PR with the suggested changes.

Arhum

**Reply by arhum-alam** [reply]
Hi Alex,

I have opened a couple of issue tickets (one is for a bug, one is a suggested feature). There were some other suggestions I had too, which seem to have been raised by other users. Happy to make a PR for either/both issues if you need.

Arhum
